mmeval/metrics/dota_map.py

```diff
@@ -0,0 +1,307 @@
+        predict_box_type (str): Box type of model results. If the QuadriBoxes
+            is used, you need to specify 'qbox'. Defaults to 'rbox'.
+        **kwargs: Keyword parameters passed to :class:`BaseMetric`.
+
+    Examples:
+
+        >>> import numpy as np
+        >>> from mmeval import DOTAMetric
+        >>> num_classes = 15
+        >>> dota_metric = DOTAMetric(num_classes=15)
+        >>>
+        >>> def _gen_bboxes(num_bboxes, img_w=256, img_h=256):
+        ...     # random generate bounding boxes in 'xywha' formart.
+        ...     x = np.random.rand(num_bboxes, ) * img_w
+        ...     y = np.random.rand(num_bboxes, ) * img_h
+        ...     w = np.random.rand(num_bboxes, ) * (img_w - x)
+        ...     h = np.random.rand(num_bboxes, ) * (img_h - y)
+        ...     a = np.random.rand(num_bboxes, ) * np.pi / 2
+        ...     return np.stack([x, y, w, h, a], axis=1)
+        >>> prediction = {
+        ...     'bboxes': _gen_bboxes(10),
+        ...     'scores': np.random.rand(10, ),
+        ...     'labels': np.random.randint(0, num_classes, size=(10, ))
+        ... }
+        >>> groundtruth = {
+        ...     'bboxes': _gen_bboxes(10),
+        ...     'labels': np.random.randint(0, num_classes, size=(10, )),
+        ...     'bboxes_ignore': _gen_bboxes(5),
+        ...     'labels_ignore': np.random.randint(0, num_classes, size=(5, ))
+        ... }
+        >>> dota_metric(predictions=[prediction, ], groundtruths=[groundtruth, ])  # doctest: +ELLIPSIS  # noqa: E501
+        {'mAP@0.5': ..., 'mAP': ...}
+    """
+
+    def __init__(self,
+                 iou_thrs: Union[float, List[float]] = 0.5,
+                 scale_ranges: Optional[List[Tuple]] = None,
+                 num_classes: Optional[int] = None,
+                 eval_mode: str = '11points',
+                 nproc: int = 4,
+                 drop_class_ap: bool = True,
+                 classwise: bool = False,
+                 predict_box_type: str = 'rbox',
+                 **kwargs) -> None:
+        super().__init__(
+            iou_thrs=iou_thrs,
+            scale_ranges=scale_ranges,
+            num_classes=num_classes,
+            eval_mode=eval_mode,
+            use_legacy_coordinate=False,
+            nproc=nproc,
+            drop_class_ap=drop_class_ap,
+            classwise=classwise,
+            **kwargs)
+        self.predict_box_type = predict_box_type
+
+    def add(self, predictions: Sequence[Dict], groundtruths: Sequence[Dict]) -> None:  # type: ignore # yapf: disable # noqa: E501
+        """Add the intermediate results to ``self._results``.
+
+        Note: The box shape of ``predictions`` and ``groundtruths`` is depends
+        on the ``self.predict_box_type``. If ``self.predict_box_type`` is
+        'rbox', the box shape should be (N, 5) which represents the (x,y,w,h,
+        angle), otherwise the box shape should be (N, 8) which represents the
+        (x1,y1,x2,y2,x3,y3,x4,y4).
```

Comment:
```suggestion
        Note:
            The box shape of ``predictions`` and ``groundtruths`` is depends
            on the ``self.predict_box_type``. If ``self.predict_box_type`` is
            'rbox', the box shape should be (N, 5) which represents the (x, y, w, h,
            angle), otherwise the box shape should be (N, 8) which represents the
            (x1, y1, x2, y2, x3, y3, x4, y4).
```